Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py

```diff
@@ -156,6 +210,53 @@ def render_PUT(self, request):
 
         parameters = http.parse_qs(request.content.read(), 1)
 
+        torrents_dir = None
+        if 'torrents_dir' in parameters and parameters['torrents_dir'] > 0:
+            torrents_dir = parameters['torrents_dir'][0]
+            if not os.path.isabs(torrents_dir):
+                request.setResponseCode(http.BAD_REQUEST)
+
+        recursive = False
+        if 'recursive' in parameters and parameters['recursive'] > 0:
+            recursive = parameters['recursive'][0]
+            if not torrents_dir:
+                request.setResponseCode(http.BAD_REQUEST)
+
+        if torrents_dir:
+            torrents_list = []
+            errors_list = []
+            filename_generator = None
+
+            if recursive:
+                def rec_gen():
+                    for root, dirnames, filenames in os.walk(torrents_dir):
+                        for f in filenames:
+                            yield os.path.join(root, f)
+
+                filename_generator = rec_gen()
+            else:
+                filename_generator = os.listdir(torrents_dir)
+
+            # Build list of .torrents to process
+            for f in filename_generator:
+                filepath = os.path.join(torrents_dir, f)
+                # TODO: add support for arbitrary encodings
+                filename = str(filepath) if sys.platform == 'win32' else filepath.decode('utf-8')
```

Comment:
As the TODO suggests, this is super dangerous. You can simply use [`sys.getdefaultencoding()`](https://docs.python.org/3/library/sys.html#sys.getdefaultencoding) instead of `'utf-8'` though.

Reply:
the problem is, in Python, filesystem encoding could be different from the default system encoding...
I'll change it, though, to `getdefaultencoding` as you suggested. It is better than what it is now.

Reply:
Oops, just your solution on my machine (`Ubuntu 18.04 en_US.UTF-8`) with some Cyrillic filenames - the thing crashed mumbling about `ascii` codec.
Apparently, Python 2 incorrectly detects system locale as `ascii`: just run
```python
sys.getdefaultencoding()
```
yourself and witness.

So, until we get a better solution, `utf-8` default encoding will be there. (Hello, `chardet` and friends!)

Reply:
Oh god, it's always ASCII: https://vstinner.github.io/painful-history-python-filesystem-encoding.html

Reply:
Oh wait, apparently I got the function wrong: it should be [`sys.getfilesystemencoding()`](https://docs.python.org/2.7/library/sys.html#sys.getfilesystemencoding)

Reply:
Looks like this last one works properly.